File: LeetCode/Medium/2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.java
class Solution {
    public String removeStars(String s) {

        int idx = 0;
        char[] arr = s.toCharArray();
        
        for(char c : arr) {
            if(c == '*') idx = Math.max(idx - 1, 0);
            else arr[idx++] = c;
        }

        return new String(arr, 0, idx);
    }
}

File: LeetCode/Medium/2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.kt
class Solution {
    fun removeStars(s: String): String {

        var idx = 0
        val arr = s.toCharArray()

        for(c in arr) {
            if(c == '*') idx = Math.max(idx - 1, 0)
            else arr[idx++] = c
        }

        return String(arr, 0, idx)
        
    }
}